characters_counter/characters_counter.py

```diff
@@ -1,16 +1,52 @@
+import collections
+import re
 from collections import Counter
 from functools import cache
+from pathlib import Path
+from typing import Dict, Union
 
 REQUIREMENT_OCCURRENCE = 1
+NUM_OF_MOST_COMMON_CHARACTERS = 3
 
 
 @cache
 def count_characters_occurrence(string: str) -> int:
     """
-    The function returns the number of characters in the string occuring only
-    once.
+    The function returns the number of characters in the string occuring only once.
     """
     characters_counts = Counter(string)
     unique_occurrences = filter(lambda occurrence: (occurrence == REQUIREMENT_OCCURRENCE), characters_counts.values())
     unique_characters_number = sum(unique_occurrences)
     return unique_characters_number
+
+
+def count_character_frequency(data_input: Union[str, Path]) -> Dict[str, int]:
```

Comment:
I didn't know this. I read that these rules, that you notised me, are since python 3.9 (if am I right), but I came across the articles even from last year which there was "an outdated" annotation in...